### ISSUE TITLE
Verifying Network IDs of the callers of received RPCs

### ISSUE THREAD

**Issue by dawkot** [godotengine/godot#7362]
There has to be a way for checking who the caller is to act accordingly.
It would be great if it could be done by using a different keyword in place of "remote" for when you want your function to only be executable if the caller has ID equal to 1.

**Comment by bojidar-bg** [comment]
More like, if you have a `move` function on the master, you want it to be callable from only one client, not all clients (as it currently is). I would go with one of the following syntaxes, though I'm not sure which is "better":
```gdscript
# GH syntax highlight fails me
master(client_id) func idea_one(param1, param2):
  if can_be_affected_by(client_id): # Checked by your own code
    # do stuff
master func idea_two(param1, param2):
  if can_be_affected_by(get_tree().get_rpc_caller_id()): # Checked by your own code
    # do stuff
```

**Comment by LeonardMeagher2** [comment]
I'm not sure of the case where this is needed.

* You could just send the unique id of the caller through the rpc function, since your function will be specifically looking for it anyways
```gdscript
master func something(id, param1, param2):
    if(id == 1):
        #do something
func call_something(p1,p2):
    rpc("something",get_tree().get_network_unique_id(),p1,p2)
```
* you could also only ever do that call from the server
```gdscript
if(get_tree().is_network_server()):
   rpc("something")
```

I personally think it's wiser to make less calls rather than filter the calls after they've been made.

**Comment by reduz** [comment]
I guess this is not needed to often, but as these are always called from
the main thread, worst case we can add some function lke
get_tree().get_rpc_caller_id()

**Comment by bojidar-bg** [comment]
@reduz Well, currently, if you have one server with 2 clients connected, a "bad" client can fully impersonate the other to the underlying server-side GDScript code, that's why it is needed.
Example:
Server scene has:
```
Node2D (master)
 - player1 - Rigidbody2D (master) [player.gd] # Or slave, but doesn't make much of a difference
 - player2 - Rigidbody2D (master) [player.gd]
```
Client1's scene has:
```
Node2D (slave)
 - player1 - Rigidbody2D (slave) [player-local-control.gd]
 - player2 - Rigidbody2D (slave) [player-remote.gd]
```
Client2's scene has:
```
Node2D (slave)
 - player1 - Rigidbody2D (slave) [player-remote.gd]
 - player2 - Rigidbody2D (slave) [player-local-control.gd]
```

Now, while the server can have to connections to each client and fully differenciate between them, it won't prevent any of them to switch the scripts around, and start controlling the other player.
So, if Client2 does `get_node("player1").set_script(preload("player-local-control.gd"))`, then he would be able to control both players at once.

**Comment by mhilbrunner** [comment]
@bojidar-bg This can be closed, as discussed here: #7853 (and merged change above)
